scionlab/forms/user_as_form.py

```diff
@@ -0,0 +1,151 @@
+# Copyright 2018 ETH Zurich
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import ipaddress
+
+from django import forms
+from django.conf import settings
+from django.core.exceptions import ValidationError
+
+from crispy_forms.helper import FormHelper
+from crispy_forms.layout import Layout, Row, Column
+from crispy_forms.bootstrap import AppendedText
+
+from scionlab.models.user_as import UserAS, AttachmentPoint
+
+
+class UserASForm(forms.ModelForm):
+    """
+    Form for UserAS creation and update.
+    """
+
+    prefix = 'userAS'
+
+    class Meta:
+        model = UserAS
+        fields = (
+            'label',
+            'installation_type',
+        )
+        labels = {
+            'label': "Label",
+        }
+        help_texts = {
+            'label': "Optional short label for your AS",
+        }
+        widgets = {
+            'installation_type': forms.RadioSelect(),
+        }
+    public_ip = forms.GenericIPAddressField(
+        help_text="Public IP for this host",
+        required=False
+    )
+    bind_ip = forms.GenericIPAddressField(
+        help_text="Bind IP for this host",
+        required=False
+    )
+
+    def __init__(self, *args, **kwargs):
+        self.user = kwargs.pop('user')
+        instance = kwargs.get('instance')
+        self.attachment_points_vpn = kwargs.pop('attachment_points_vpn', None)
+        initial = kwargs.pop('initial', {})
+        if instance:
+            initial['public_ip'] = instance.host.public_ip
+            initial['bind_ip'] = instance.host.bind_ip
+        #  self.helper = self._crispy_helper()
```

Comment:
The reason I ditched the crispy Layout is because it generates a `Form` with the `<form></form>` included, and we need to insert two different `Form`s in a single request, namely: `UserASForm` and `AttachmentLinkForm`. Nonetheless, the display of the errors still works good.

Anyway, I think I found in the docs that it is possible to instruct the helper not to create the `<form></form>` tag and the `csrf_token`, so I believe I can use the `Layout` feature as well :ok_hand: